runtime/autoload/lsp/completion.vim

```diff
@@ -0,0 +1,43 @@
+let s:last_location = -1
+
+""
+" Omni completion with LSP
+function! lsp#completion#omni(findstart, base) abort
+  " If we haven't started, then don't return anything useful
+  if !luaeval("require('lsp.api').plugin.client_has_started(_A)", &filetype)
+    return a:findstart ? -1 : []
+  endif
+
+  if a:findstart == 1
+    let s:last_location =  col('.')
+
+    let line_to_cursor = strpart(getline('.'), 0, col('.') - 1)
+    let [string_result, start_position, end_position] = matchstrpos(line_to_cursor, '\k\+$')
+    let length = end_position - start_position
+
+    return len(line_to_cursor) - length
+  elseif a:findstart == 0
+    let params = luaeval("require('lsp.api').structures.CompletionParams("
```

Comment:
Rather than invoking multiple lua wrappers in sequence and needlessly convert data back and forth (which risks being lossy), we should just call one lua function that does what is needed ( `requests.textDocument.completion` maybe ? though the relation between the structures and requests modules looks a bit weird, I need to figure that out )